CesiumGltfContent/src/GltfUtilities.cpp

```diff
@@ -901,4 +1228,265 @@ void GltfUtilities::compactBuffer(
+       &primitive,
+       cullBackFaces,
+       &tClosest,
+       &warnings](const auto& positionView) {
+        // Bail on invalid view
+        if (positionView.status() != AccessorViewStatus::Valid) {
+          warnings.push_back(
+              "Skipping mesh with an invalid position component type");
+          return;
+        }
+
+        bool hasIndexedTriangles = primitive.indices != -1;
+        if (hasIndexedTriangles) {
+          assert(primitive.indices >= 0);
+          const Accessor& indexAccessor =
+              model.accessors[static_cast<size_t>(primitive.indices)];
+
+          // Ignore float index types, these are invalid
+          // From the glTF spec...
+          // "Indices MUST be non-negative integer numbers."
+          if (indexAccessor.componentType == Accessor::ComponentType::FLOAT) {
+            warnings.push_back(
+                "Skipping mesh with an invalid index component type");
+            return;
+          }
+
+          createAccessorView(
+              model,
+              indexAccessor,
+              [&transformedRay,
+               &positionView,
+               &primitive,
+               cullBackFaces,
+               &tClosest,
+               &warnings](const auto& indexView) {
+                // Bail on invalid view
+                if (indexView.status() != AccessorViewStatus::Valid) {
+                  warnings.push_back(
+                      "Could not create accessor view for mesh indices");
+                  return;
+                }
+
+                findClosestIndexedRayHit<
+                    decltype(positionView),
+                    decltype(indexView)>(
+                    transformedRay,
+                    positionView,
+                    indexView,
+                    primitive,
+                    cullBackFaces,
+                    tClosest,
+                    warnings);
+              });
+        } else {
+          // Non-indexed triangles
+          findClosestRayHit<decltype(positionView)>(
+              transformedRay,
+              positionView,
+              primitive,
+              cullBackFaces,
+              tClosest,
+              warnings);
+        }
+      });
+  assert(tClosest >= -1);
+
+  if (tClosest == -1)
+    return std::optional<glm::dvec3>();
+
+  // It's temping to return the t value to the caller, but each primitive might
+  // have different matrix transforms with different scaling values. The caller
+  // should instead compare world distances.
+  return transformedRay.pointFromDistance(tClosest);
+}
+
+GltfUtilities::IntersectResult GltfUtilities::intersectRayGltfModel(
+    const CesiumGeometry::Ray& ray,
+    const CesiumGltf::Model& gltf,
+    bool cullBackFaces,
+    const glm::dmat4x4& gltfTransform) {
+  glm::dmat4x4 rootTransform = applyRtcCenter(gltf, gltfTransform);
+  rootTransform = applyGltfUpAxisTransform(gltf, rootTransform);
+
+  IntersectResult result;
+  IntersectResult::Hit closestHit;
```

Comment:
Minor thing, but it seems like it would be cleaner to capture `&result` in the lambda below, and then use the hit embedded in it. That avoids the need to compare `rayToWorldPointDistanceSq` to -1.0 to see if there's already a hit (instead you can use the optional), and it avoids the need to move the hit into the actual result at the end.

Reply:
I like it. Changes committed